presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/TestCachingHiveMetastore.java

```diff
@@ -209,8 +224,8 @@ public void testCachingWithPartitionVersioning()
         assertEquals(mockClient.getAccessCount(), 3);
 
         // This call should invalidate the partition cache because when partition versioning is enabled
-        // partitions with older or empty version will be deleted from cache. And the hivePartitionVersionFetcher used in creating
-        // partitionCachingEnabledmetastore will set the version as Optional.empty() for all the partitions fetched.
+        // partitions with empty or different version will be deleted from cache. And the MockPartitionMutator used in creating
+        // partitionCachingEnabledmetastore will set the version as random integer for all the partitions fetched.
```

Comment:
Let's test for equal, larger and smaller for this? This should give a full coverage and make the test more deterministic.